cmd/create/machinepool/cmd.go

```diff
@@ -17,255 +17,100 @@ limitations under the License.
-	)
-
-	flags.StringVar(
-		&args.SpotMaxPrice,
-		"spot-max-price",
-		"on-demand",
-		"Max price for spot instance. If empty use the on-demand price.",
-	)
-
-	flags.BoolVar(
-		&args.MultiAvailabilityZone,
-		"multi-availability-zone",
-		true,
-		"Create a multi-AZ machine pool for a multi-AZ cluster")
-
-	flags.StringVar(
-		&args.AvailabilityZone,
-		"availability-zone",
-		"",
-		"Select availability zone to create a single AZ machine pool for a multi-AZ cluster")
-
-	flags.StringVar(
-		&args.Subnet,
-		"subnet",
-		"",
-		"Select subnet to create a single AZ machine pool for BYOVPC cluster")
-
-	flags.StringVar(
-		&args.Version,
-		"version",
-		"",
-		"Version of OpenShift that will be used to install a machine pool for a hosted cluster,"+
-			" for example \"4.12.4\"",
-	)
+type CreateMachinePool struct {
+	service machinepool.MachinePoolService
+}
 
-	flags.BoolVar(
-		&args.Autorepair,
-		"autorepair",
-		true,
-		"Select auto-repair behaviour for a machinepool in a hosted cluster.",
-	)
+func NewCreateMachinePool(spec CreateMachinePoolSpec) CreateMachinePool {
+	return CreateMachinePool{
+		service: spec.Service,
+	}
+}
 
-	flags.StringVar(
-		&args.TuningConfigs,
-		"tuning-configs",
-		"",
-		"Name of the tuning configs to be applied to the machine pool. Format should be a comma-separated list. "+
-			"Tuning config must already exist. "+
-			"This list will overwrite any modifications made to node tuning configs on an ongoing basis.",
-	)
+func NewCreateMachinePoolCommand() *cobra.Command {
+	cmd, options := mpOpts.BuildMachinePoolCreateCommandWithOptions()
+	cmd.Run = rosa.DefaultRunner(rosa.RuntimeWithOCM(), CreateMachinepoolRunner(options))
+	return cmd
+}
 
-	flags.StringVar(
-		&args.KubeletConfigs,
-		"kubelet-configs",
-		"",
-		"Name of the kubelet config to be applied to the machine pool. A single kubelet config is allowed. "+
-			"Kubelet config must already exist. "+
-			"This will overwrite any modifications made to node kubelet configs on an ongoing basis.",
-	)
+// Create machine pool based on cluster type
+func (m CreateMachinePool) createMachinePoolBasedOnClusterType(r *rosa.Runtime,
+	cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster,
+	options *mpOpts.CreateMachinepoolUserOptions) error {
+	if cluster.Hypershift().Enabled() {
+		return m.service.CreateNodePools(r, cmd, clusterKey, cluster, options)
+	}
+	return m.service.CreateMachinePool(r, cmd, clusterKey, cluster, options)
+}
 
-	flags.StringVar(&args.RootDiskSize,
-		"disk-size",
-		"",
-		"Root disk size with a suffix like GiB or TiB",
-	)
+// Original function refactored to use the new helper functions
+func CreateMachinepoolRunner(userOptions *mpOpts.CreateMachinepoolUserOptions) rosa.CommandRunner {
+	return func(ctx context.Context, r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+		var err error
+		options := NewCreateMachinepoolOptions()
+		clusterKey := r.GetClusterKey()
+		options.args = userOptions
```

Comment:
@den-rgb Should this be calling the `Bind` method on the `options` instance?

Reply:
The Bind i left in was accidental, the Bind in this instance is redundant because argument order doesnt have to be sequential and the Binds for other methods expect the name of the machinepool to be first, whereas create doesn't